libs/go-kibana-rest/kbapi/api.kibana_synthetics.go

```diff
@@ -0,0 +1,323 @@
+package kbapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/go-resty/resty/v2"
+	log "github.com/sirupsen/logrus"
+	"time"
+)
+
+const (
+	basePathKibanaSynthetics = "/api/synthetics" // Base URL to access on Kibana save objects
+	privateLocationsSuffix   = "/private_locations"
+	monitorsSuffix           = "/monitors"
+)
+
+type MonitorID string
+type MonitorType string
+type MonitorLocation string
+type MonitorSchedule int
+type HttpMonitorMode string
+
+type KibanaError struct {
+	Code    int    `json:"statusCode,omitempty"`
+	Error   string `json:"error,omitempty"`
+	Message string `json:"message,omitempty"`
+}
+
+type JsonObject map[string]interface{}
+
+const (
+	Http    MonitorType = "http"
+	Tcp     MonitorType = "tcp"
+	Icmp    MonitorType = "icmp"
+	Browser MonitorType = "browser"
+
+	Every1Minute    MonitorSchedule = 1
+	Every2Minutes                   = 2
+	Every3Minutes                   = 3
+	Every5Minutes                   = 5
+	Every10Minutes                  = 10
+	Every15Minutes                  = 15
+	Every20Minutes                  = 20
+	Every30Minutes                  = 30
+	Every60Minutes                  = 60
+	Every120Minutes                 = 120
+	Every240Minutes                 = 240
+
+	Japan         MonitorLocation = "japan"
+	India                         = "india"
+	Singapore                     = "singapore"
+	AustraliaEast                 = "australia_east"
+	UnitedKingdom                 = "united_kingdom"
+	Germany                       = "germany"
+	CanadaEast                    = "canada_east"
+	Brazil                        = "brazil"
+	USEast                        = "us_east"
+	USWest                        = "us_west"
+
+	ModeAll HttpMonitorMode = "all"
+	ModeAny                 = "any"
+)
+
+type KibanaSyntheticsMonitorAPI struct {
+	Add    KibanaSyntheticsMonitorAdd
+	Delete KibanaSyntheticsMonitorDelete
+	Get    KibanaSyntheticsMonitorGet
+}
+
+type KibanaSyntheticsPrivateLocationAPI struct {
+	Create KibanaSyntheticsPrivateLocationCreate
+	Delete KibanaSyntheticsPrivateLocationDelete
+	Get    KibanaSyntheticsPrivateLocationGet
+}
+
+type MonitorAlertConfig struct {
+	Status struct {
```

Comment:
Do we ever need to create these structs in code or will they only be used for serialisation? I'm not opposed to the anonymous struct, but if we expect to reference them in code it might be easier to define them as their own type.

Reply:
That is a good question. tbh: I don't know yet, how will customer use this API. I expect to refactor this part after adding terraform changes. 

For scope of this PR: I have moved nested types to separate structures.